src/go-slang/lib/heap/index.ts

```diff
@@ -359,6 +372,15 @@ export class Heap {
     this.memory.setUint8(ptr_heap_addr + 3, 0) // initialize buffer size to 0
     return ptr_heap_addr
   }
+  /* Memory Layout of a WaitGroup: 
+   * [0-7:tag][0-7:count] (2 words) 
+   */
+  public allocateWaitGroup(): HeapAddress {
+    const ptr_heap_addr = this.allocateTaggedPtr(PointerTag.WaitGroup, 2)
```

Comment:
```suggestion
    const ptr_heap_addr = this.allocateTaggedPtr(PointerTag.WaitGroup, 1)
```

The `allocateTaggedPtr` takes in the size of the underlying data structure, so it implicitly +1 to account for the size of the tag heap object

Reply:
Reasonable.